### PR TITLE
refactor cleaning ethograms to use keyboard events

### PR DESCRIPTION
**Pull request by clewis7**

### CODE REVIEW

**Comment by clewis7** [comment]
@kushalkolar thoughts?

File: animal_soup/_ethogram_cleaner.py [hunk 1]
```diff
@@ -10,6 +10,9 @@
 from typing import Union
 from .batch_utils import validate_path
 
+HIGHLIGHT_GRAPHICS = ["lift_highlight", "handopen_highlight", "grab_highlight", "sup_highlight", "atmouth_highlight",
```
Comment (kushalkolar) [inline animal_soup/_ethogram_cleaner.py:13]: one per line is easier to read

File: animal_soup/_ethogram_cleaner.py [hunk 2]
```diff
@@ -25,12 +28,11 @@ def __init__(
         if clean_df is None:
             df_dir, relative_path = self._dataframe.paths.split(self._dataframe.paths.get_df_path())
             self.clean_df_path = df_dir.joinpath(relative_path.stem).with_name(f'{relative_path.stem}_cleaned').with_suffix('.hdf')
-            # check if clean_df exists and user forgot to pass
+            # check if clean_df exists and user didn't
```
Comment (kushalkolar) [inline animal_soup/_ethogram_cleaner.py:31]: and user didn't what?

File: animal_soup/_ethogram_cleaner.py [hunk 3]
```diff
@@ -10,6 +10,9 @@
 from typing import Union
 from .batch_utils import validate_path
 
+HIGHLIGHT_GRAPHICS = ["lift_highlight", "handopen_highlight", "grab_highlight", "sup_highlight", "atmouth_highlight",
+                   "chew_highlight"]
+
 class EthogramCleaner(EthogramVizContainer):
```
Comment (kushalkolar) [inline animal_soup/_ethogram_cleaner.py:16]: docstring

Comment (kushalkolar) [reply animal_soup/_ethogram_cleaner.py:16]: do you have pycharm setup to make it automatically?

File: animal_soup/_ethogram_cleaner.py [hunk 4]
```diff
@@ -132,28 +113,47 @@ def ethogram_selection_event_handler(self, ev):
...
-                self.behavior_buttons.value]
-
-    def save_ethogram(self, obj):
+    def ethogram_key_event_handler(self, obj):
+        """Event handler for handling keyboard events to clean up ethograms."""
+        current_highlight_name = self.currently_selected_behavior.name
+        current_ix = HIGHLIGHT_GRAPHICS.index(current_highlight_name)
+        current_behavior_name = self.currently_selected_behavior.name.split('_')[0]
+        selected_ixs = self.plot.selectors[0].get_selected_indices(self.plot[current_behavior_name])
+        if obj.key == 's':
```
Comment (kushalkolar) [inline animal_soup/_ethogram_cleaner.py:122]: comments for every if else claus, this is unreadable

File: animal_soup/_ethogram_cleaner.py [hunk 5]
```diff
@@ -132,28 +113,47 @@ def ethogram_selection_event_handler(self, ev):
...
-        else:
-            self.plot.graphics[behavior_ix].colors[selected_ixs[0]:selected_ixs[-1]] = ETHOGRAM_COLORS[
-                self.behavior_buttons.value]
-
-    def save_ethogram(self, obj):
+    def ethogram_key_event_handler(self, obj):
+        """Event handler for handling keyboard events to clean up ethograms."""
+        current_highlight_name = self.currently_selected_behavior.name
+        current_ix = HIGHLIGHT_GRAPHICS.index(current_highlight_name)
+        current_behavior_name = self.currently_selected_behavior.name.split('_')[0]
```
Comment (kushalkolar) [inline animal_soup/_ethogram_cleaner.py:120]: This is so much easier if you make a `@property` for `selected_behavior`.

File: animal_soup/_ethogram_cleaner.py (hunk 4 above)
Comment (kushalkolar) [reply animal_soup/_ethogram_cleaner.py:122]: I'm really having a hard time parsing out this entire if else block

File: animal_soup/_ethogram_cleaner.py [hunk 6]
```diff
@@ -163,26 +163,11 @@ def save_ethogram(self, obj):
         # save clean_df to disk
         self._clean_df.to_hdf(self.clean_df_path, key='df')
 
-    def reset_ethogram(self, obj):
+    def reset_ethogram(self):
```
Comment (kushalkolar) [inline animal_soup/_ethogram_cleaner.py:166]: do you want to have the option to reset only for the current behavior vs. entire ethogram?

**Review by kushalkolar — CHANGES_REQUESTED** [review]

File: animal_soup/_ethogram_cleaner.py (hunk 6 above)
Comment (clewis7) [reply animal_soup/_ethogram_cleaner.py:166]: yeah, I think that would be a good idea, will add that in

**Pull request merged**